spec/controllers/shipments_controller_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.describe ShipmentsController, type: :controller do
+  let(:successful_params) do
+    { origin: { "country" => "US", "city" => "Seattle", "state" => "WA", "zip" => 98105 },
+    destination: { country: "US", city: "Seattle", "state" => "WA", "zip" => 98105 },
```

Comment:
Another style comment: the keys in this hash are inconsistent -- some being symbols and others strings. I prefer not to use the hash rocket (`=>`) wherever possible, so my style would be to have all symbols for keys here. It will vary depending on your team's preferences and standards, but it's good practice to use consistent syntax.